src/core/providers/ogr/qgsogrlayermetadataprovider.cpp

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************
+  qgsogrlayermetadataprovider.cpp - QgsOgrLayerMetadataProvider
+
+ ---------------------
+ begin                : 24.8.2022
+ copyright            : (C) 2022 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsogrlayermetadataprovider.h"
+#include "qgsprovidermetadata.h"
+#include "qgsproviderregistry.h"
+#include "qgsfeedback.h"
+
+
+QString QgsOgrLayerMetadataProvider::type() const
+{
+  return QStringLiteral( "ogr" );
+}
+
+QgsLayerMetadataSearchResult QgsOgrLayerMetadataProvider::search( const QString &searchString, const QgsRectangle &geographicExtent, QgsFeedback *feedback ) const
+{
+  QgsLayerMetadataSearchResult results;
```

Comment:
This whole class is basically identical to the postgres layer metadata provider class. We could avoid this by just making this class generic and then constructing it with a type argument set to "ogr" or "postgres".

Reply:
My first attempt was exactly like you suggest but I had to make a custom implementation for pg rasters, then I was undecided if (following the principle of least surprise) it would have been better to move the implementation away from the abstract class and duplicate it for ogr and pg or leave it and reimplement it for pg raster only. I chose the first option but I'm happy to go back. My other rationale was that as an interface it was more logic to have it all pure but I'm not religious about that.

Reply:
Now PG raster is gone but there is still a blocker to move the implementation into the abstract base class: OGR does not have the flag to enable/disable the md storage.

Reply:
It's not a blocker to me -- go with whatever you feel is the cleanest approach, you know this code best :+1: